### PR TITLE
HLM-5926 changing base color

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreview.js [hunk 1]
```diff
@@ -126,7 +126,7 @@ const MicroplanPreview = ({
       setOperatorsObject(temp.ruleConfigureOperators);
     }
     if (microplanPreviewAggregatesList) setMicroplaPreviewAggregates(microplanPreviewAggregatesList);
-  }, [state?.UIConfiguration, state?.Schemas, state?.Resources, state?.MicroplanPreviewAggregates]);
+  }, []);
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreview.js:129]: I dont want to render it what those things change

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Upload.js [hunk 2]
```diff
@@ -13,8 +13,8 @@ import { SpatialDataPropertyMapping } from "./resourceMapping";
 import shp from "shpjs";
 import { JsonPreviewInExcelForm } from "./JsonPreviewInExcelForm";
 import { ButtonType1, ButtonType2, CloseButton, ModalHeading } from "./CommonComponents";
-import { Loader, Modal, Toast } from "@egovernments/digit-ui-components";
-import { EXCEL, GEOJSON, LOCALITY, SHAPEFILE } from "../configs/constants";
+import { Loader, Toast } from "@egovernments/digit-ui-components";
+import { EXCEL, GEOJSON, LOCALITY, PRIMARY_THEME_COLOR, SHAPEFILE } from "../configs/constants";
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Upload.js:17]: I am using it

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Hypothesis.js [hunk 3]
```diff
@@ -44,21 +48,27 @@ const Hypothesis = ({ campaignType = "SMC", microplanData, setMicroplanData, che
       setAssumptions(microplanData.hypothesis);
     }
 
+    fetchDataAndUpdateState();
+  }, []);
+
+  const fetchDataAndUpdateState = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Hypothesis.js:54]: I dont want to render it what those things change

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/processHierarchyAndData.js [hunk 4]
```diff
@@ -52,12 +53,14 @@ export const processHierarchyAndData = (hierarchy, allData) => {
...
             }
           }
-
+          parent = cellValue;
           currentNode = currentNode[cellValue].children;
         });
       });
 
       // Merge dataHierarchicalData into hierarchicalData
+      debugger
```
Comment (nipunarora-eGov) [inline micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/processHierarchyAndData.js:62]: @siddhant-nawale-egov remove debugger

File: micro-ui/web/package.json [hunk 5]
```diff
@@ -14,7 +14,7 @@
   ],
   "homepage": "/digit-ui",
   "dependencies": {
-    "@egovernments/digit-ui-libraries": "1.8.0-beta.2",
+    "@egovernments/digit-ui-libraries": "1.8.1-beta.1",
```
Comment (nipunarora-eGov) [inline micro-ui/web/package.json:17]: this update is required?

Comment (siddhant-nawale-egov) [reply micro-ui/web/package.json:17]: We am using that in local

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Upload.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Upload.js:17]: They are component from different libraries

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Hypothesis.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Hypothesis.js:54]: I dont want it to rerender

**Review by nipunarora-eGov — APPROVED** [review]

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/processHierarchyAndData.js [hunk 6]
```diff
@@ -80,21 +81,43 @@ export const processHierarchyAndData = (hierarchy, allData) => {
...
+        else value.data.children = data1[key].data.children;
       data1[key].data = value.data; // Merge data
+      if (value.children) {
+        if (Array.isArray(data1[key].data.children))
+          data1[key].data.children = [...data1[key].data.children, ...Object.keys(value.children)].join(", ");
+        else data1[key].data.children = Object.keys(value.children).join(", ");
+      } else data1[key].data.children = null;
+      if(data1[key].data.children?.length === 0)
+        data1[key].data.children = null
+
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/processHierarchyAndData.js:102]: Suggestion doesnt work

**Pull request merged**